### DISCUSSION TITLE
trignometric visualizer

### DISCUSSION THREAD

**Discussion by sfederici** [leopard-js/leopard-mentors#8] [Cleanup Crew]
### Scratch Project URL

https://scratch.mit.edu/projects/927737037

### Leopard Project URL

https://leopardjs.com/projects/37tFt1S4te

### What kind of error(s) are you experiencing?

- [ ] Leopard is failing to generate JavaScript code
- [ ] The JavaScript code generated by Leopard doesn't work correctly
- [ ] The JavaScript code generated by Leopard is confusing or needs improvement
- [X] Other

### Description of issue(s)

The background in the Leopard translation is white, not black. So the white text is not visible

**Comment by PullJosh** [comment]
I can't access your Leopard project because it isn't shared. When I tried converting the project myself (just now) the background appears to be black, correctly.

However, the text is still not visible.

**Comment by PullJosh** [comment]
Not to pat ourselves on the back, but I'm quite pleased that this hacked block in the project translated seemingly-correctly:
<img width="377" alt="image" src="https://github.com/user-attachments/assets/30340e12-8bdf-4c12-8944-7559c0cd7710">

**Reply by PullJosh** [reply]
Well, nearly, anyway. It turns out that within the Function Buttons sprite there is the following custom block:
<img width="386" alt="image" src="https://github.com/user-attachments/assets/cbad6e98-e702-4aeb-bc61-1db8c2dbfaee">

Notice that this block runs the above block _within_ a "for each" loop, creating two nested "for each" loops that both use the same variable, `i`. The problem is that the inner loop mutates `i` and prevents the outer loop from running the same way that Scratch does. (In Scratch, the loops do not interfere with each other at all.)

This is a pretty esoteric issue. We could try to fix it in Leopard, but I don't think it's a high priority since this is a hacked block. Rewriting these loops to use non-hacked blocks (or changing the JS for loops to use a local variable rather than global) fixes the issue.

**Comment by PullJosh** [comment]
Aha! 🔍🕵️ There's one more Leopard bug that is causing issues. In Scratch, a hidden sprite (that used the "hide" block) will still appear when using the "stamp" pen block. In Leopard, using "stamp" with a hidden sprite doesn't work.

This particular project can be fixed by showing the sprite:
<img width="531" alt="image" src="https://github.com/user-attachments/assets/7e92a954-87e4-48bc-9e04-b58dd1418ee4">

But of course we should fix this behavior in general (https://github.com/leopard-js/leopard/issues/205).

**Fixed version of this project:** https://leopardjs.com/projects/VFLg_A0nge

**Comment by sfederici** [comment]
I'm reopening as, in the [r,-r] graph, the dots are red in Scratch but they are black in Leopard
